__tests__/util/poll.test.js

```diff
@@ -0,0 +1,33 @@
+import poll, { TimeoutError } from '../../app/util/poll'
+
+describe('poll functionality', () => {
+  test('should stop when condition fulfilled', async () => {
+    // 10 rounds of 10ms
+    const pollCfg = { attempts: 10, frequency: 10 }
+
+    // pulfill the poll condition after 5th attempt
```

Comment:
Not a big deal but believe you meant `fulfill` here 😄

Reply:
LOL It's a new term I coined apparently - poll fulfill => pulfill